substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java

```diff
@@ -100,6 +98,11 @@ protected SystemPropertiesSupport() {
         initializeProperty("java.ext.dirs", "");
         initializeProperty("java.library.path", "");
 
+        String targetName = System.getProperty("svm.targetName");
```

Comment:
This is quite loosely coupled (i.e. allows to lie about `os.name` and  `os.arch` independent from actually doing cross-compilation). Is this a temporary solution? (I.e. the setting of `os.name` and  `os.arch` should be driven by actual cross-compilation options.)

Reply:
svm.targetArch is already used for actual cross-compilation, https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/macho/MachOObjectFile.java#L83

Reply:
Hmm, maybe I miss something but that seems to just tweak how relocations are generated in the output object file. Would it make more sense to derive `os.arch` from the `org.graalvm.nativeimage.Platform` we are building for?

Reply:
I.e. something like
```
if (Platform.includedIn(Platform.AArch64.class)) {
   ...
```